### ISSUE TITLE
`gochecknoglobals` underreports total global variables

### ISSUE THREAD

**Issue by wildan3105** [leighmcculloch/gochecknoglobals#50]
Hi there,
I want to report a possible issue. When running `golangci-lint` with `gochecknoglobals` enabled, the reported count of global variables appears incomplete

**Details**

Running the following command:
```bash
golangci-lint run ./...
```
produces:

```bash
192 issues:
* errcheck: 35
* gochecknoglobals: 50
* ineffassign: 7
* staticcheck: 50
* unused: 50
```

However, there are **at least 70** global variables defined in the repository, meaning `gochecknoglobals` reports fewer issues than expected. 

**Environment**:

```bash
# golang-ci version
% golangci-lint --version
golangci-lint has version 2.5.0 built with go1.25.1 from ff63786c on 2025-09-21T19:04:05Z

# go version
% go version
go version go1.25.2 darwin/amd64
```

**Configuration**:
```yaml
version: "2"

linters:
  enable:
    - gochecknoglobals # checks for usage of global variables
```

**Expected behaviour**
`gochecknoglobals` should report all detected global variables in the project.

**Comment by leighmcculloch** [comment]
Thanks for reporting. Could you share a link to the codebase and commit sha that this occurred on? Or alternatively open a PR with a test that fails with your case? Or alternatively a list of the globals not discovered?

**Comment by wildan3105** [comment]
Hmm, actually the codebase is kind of confidential so I can't disclose the details of it. But let me quickly reproduce this in arbitrary repository which doesn't contain sensitive logic/information but still demonstrate the "underreport".

**Comment by wildan3105** [comment]
Hi @leighmcculloch, I was able to reproduce the "underreport" in my public repository. Details:
- var detected: https://github.com/search?q=repo%3Awildan3105%2Fconverto%20var&type=code (at least 13 global variables found)
- action details: https://github.com/wildan3105/converto/actions/runs/19129695726/job/54667201043 (only detect 11)

Let me know if you need more info. Thanks for replying!

**Comment by wildan3105** [comment]
Adding screenshot of the github action output in case you can't see it:

<img width="1770" height="867" alt="Image" src="https://github.com/user-attachments/assets/2acca2d5-4f03-452a-8fdc-b3ab3b5807b8" />

**Comment by wildan3105** [comment]
Ok after checking in-depth, this rule only missed a single global variable which was defined here:

https://github.com/wildan3105/converto/blob/13d96763f5b4b30b67fa8396c56874e3a6e4c8c1/pkg/handler/health.go#L21

Everything else was accurately detected.

**Comment by wildan3105** [comment]
Sorry! False alarm. Turns out, I had to set `--max-same-issues=0` to show duplicate variable/message. Thanks and sorry for the inconvenience!

**Comment by wildan3105** [comment]
Here's for log for the reference when running golangci-lint without setting the `max-same-issues` flag:

```bash
INFO [runner/filename_unadjuster] Pre-built 0 adjustments in 1.434875ms 
INFO [linters_context/goanalysis] analyzers took 0s with no stages 
INFO [runner/max_same_issues] 5/8 issues with text "log is a global variable" were hidden, use --max-same-issues 
INFO [runner] Issues before processing: 28, after processing: 23 

```

**Comment by leighmcculloch** [comment]
Ah, glad the problem is resolved!
